src/details/ArborX_ExperimentalTreeHelpers.hpp

```diff
@@ -0,0 +1,109 @@
+/****************************************************************************
+ * Copyright (c) 2017-2021 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_DETAILS_HDBSCAN_HPP
+#define ARBORX_DETAILS_HDBSCAN_HPP
+
+#include <ArborX_DetailsHappyTreeFriends.hpp>
+#include <ArborX_LinearBVH.hpp>
+
+namespace ArborX
+{
+
+namespace Experimental
+{
+
+template <class ExecutionSpace, class BVH, class LabelsIn, class LabelsOut>
+void initBVHLabels(ExecutionSpace const &exec_space, BVH const &bvh,
```

Comment:
```suggestion
void initLabels(ExecutionSpace const &exec_space, BVH const &bvh,
```

Reply:
The reason I added `BVH` to the names of the functions is because the results depend on the BVH implementation. In other words, if I try to use other search indices, it may not be the same. For that reason, I think using a general `initLabels` and such is not appropriate.